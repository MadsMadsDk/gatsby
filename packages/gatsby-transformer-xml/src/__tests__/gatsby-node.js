const Promise = require(`bluebird`)

const { onCreateNode } = require(`../gatsby-node`)
describe(`Process XML nodes correctly`, () => {
  const node = {
    name: `nodeName`,
    id: `whatever`,
    parent: null,
    children: [],
    internal: {
      contentDigest: `whatever`,
      mediaType: `application/xml`,
      name: `test`,
    },
  }

  // Make some fake functions its expecting.
  node.content = `
      <?xml version="1.0"?>
      <catalog>
        <book id="bk101">
            <author>Gambardella, Matthew</author>
            <title>XML Developer's Guide</title>
            <genre>Computer</genre>
            <price>44.95</price>
            <publish_date>2000-10-01</publish_date>
            <publish_date_explicit>
              <year>2000</year>
              <month>
                10
                <month_name lang="en-gb">October</month_name>
              </month>
              <day>01</day>
            </publish_date_explicit>
            <description>An in-depth look at creating applications
            with XML.</description>
            <description_html><![CDATA[
              <p>An in-depth look at creating applications
            with XML.</p>
            ]]></description_html>
         </book>
         <book id="bk102">
            <author>Ralls, Kim</author>
            <title>Midnight Rain</title>
            <genre>Fantasy</genre>
            <price>5.95</price>
            <publish_date>2000-12-16</publish_date>
            <publish_date_explicit>
              <year>2000</year>
              <month>
                12
                <month_name lang="en-gb">December</month_name>
              </month>
              <day>16</day>
            </publish_date_explicit>
            <description>A former architect battles corporate zombies,
            an evil sorceress, and her own childhood to become queen
            of the world.</description>
            <description_html><![CDATA[
              <p>A former architect battles <strong>corporate zombies</strong>,
            an evil sorceress, and her own childhood to become queen
            of the world.</p>
            ]]></description_html>
         </book>
      </catalog>
    `
  const loadNodeContent = node => Promise.resolve(node.content)

  it(`correctly creates nodes from XML`, async () => {
    const createNode = jest.fn()
    const createParentChildLink = jest.fn()
    const actions = { createNode, createParentChildLink }
    const createNodeId = jest.fn()
    createNodeId.mockReturnValue(`uuid-from-gatsby`)
    const createContentDigest = jest.fn().mockReturnValue(`contentDigest`)

    await onCreateNode(
      {
        node,
        loadNodeContent,
        actions,
        createNodeId,
        createContentDigest,
      },
      {}
    ).then(() => {
      expect(createNode.mock.calls).toMatchSnapshot()
      expect(createParentChildLink.mock.calls).toMatchSnapshot()
      expect(createNode).toHaveBeenCalledTimes(2)
      expect(createParentChildLink).toHaveBeenCalledTimes(2)
    })
  })

  it(`correctly creates nodes from XML with plugin options set`, async () => {
    const createNode = jest.fn()
    const createParentChildLink = jest.fn()
    const actions = { createNode, createParentChildLink }
    const createNodeId = jest.fn()
    createNodeId.mockReturnValue(`uuid-from-gatsby`)
    const createContentDigest = jest.fn().mockReturnValue(`contentDigest`)

    await onCreateNode(
      {
        node,
        loadNodeContent,
        actions,
        createNodeId,
        createContentDigest,
      },
      { ignoreDeclaration: false }
    ).then(() => {
      expect(createNode.mock.calls).toMatchSnapshot()
      expect(createParentChildLink.mock.calls).toMatchSnapshot()
      expect(createNode).toHaveBeenCalledTimes(3)
      expect(createParentChildLink).toHaveBeenCalledTimes(3)
    })
  })

  it(`correctly creates nodes from XML in the legacy v2.x format`, async () => {
    const createNode = jest.fn()
    const createParentChildLink = jest.fn()
    const actions = { createNode, createParentChildLink }
    const createNodeId = jest.fn()
    createNodeId.mockReturnValue(`uuid-from-gatsby`)
    const createContentDigest = jest.fn().mockReturnValue(`contentDigest`)

    await onCreateNode(
      {
        node,
        loadNodeContent,
        actions,
        createNodeId,
        createContentDigest,
      },
      { useLegacyOutput: true, ignoreDeclaration: false }
    ).then(() => {
      expect(createNode.mock.calls).toMatchSnapshot()
      expect(createParentChildLink.mock.calls).toMatchSnapshot()
      expect(createNode).toHaveBeenCalledTimes(3)
      expect(createParentChildLink).toHaveBeenCalledTimes(3)
    })
  })
})
