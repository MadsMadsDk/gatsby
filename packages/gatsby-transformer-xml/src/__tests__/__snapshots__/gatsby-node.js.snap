// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Process XML nodes correctly correctly creates nodes from XML 1`] = `
Array [
  Array [
    Object {
      "attributes": Object {
        "id": "bk101",
      },
      "author": Object {
        "text": "Gambardella, Matthew",
      },
      "children": Array [],
      "description": Object {
        "text": "An in-depth look at creating applications
            with XML.",
      },
      "description_html": Object {
        "cdata": "<p>An in-depth look at creating applications
            with XML.</p>",
      },
      "genre": Object {
        "text": "Computer",
      },
      "id": "bk101",
      "internal": Object {
        "contentDigest": "contentDigest",
        "type": "NodeNameXml",
      },
      "parent": "whatever",
      "price": Object {
        "text": "44.95",
      },
      "publish_date": Object {
        "text": "2000-10-01",
      },
      "title": Object {
        "text": "XML Developer's Guide",
      },
    },
  ],
  Array [
    Object {
      "attributes": Object {
        "id": "bk102",
      },
      "author": Object {
        "text": "Ralls, Kim",
      },
      "children": Array [],
      "description": Object {
        "text": "A former architect battles corporate zombies,
            an evil sorceress, and her own childhood to become queen
            of the world.",
      },
      "description_html": Object {
        "cdata": "<p>A former architect battles <strong>corporate zombies</strong>,
            an evil sorceress, and her own childhood to become queen
            of the world.</p>",
      },
      "genre": Object {
        "text": "Fantasy",
      },
      "id": "bk102",
      "internal": Object {
        "contentDigest": "contentDigest",
        "type": "NodeNameXml",
      },
      "parent": "whatever",
      "price": Object {
        "text": "5.95",
      },
      "publish_date": Object {
        "text": "2000-12-16",
      },
      "title": Object {
        "text": "Midnight Rain",
      },
    },
  ],
]
`;

exports[`Process XML nodes correctly correctly creates nodes from XML 2`] = `
Array [
  Array [
    Object {
      "child": Object {
        "attributes": Object {
          "id": "bk101",
        },
        "author": Object {
          "text": "Gambardella, Matthew",
        },
        "children": Array [],
        "description": Object {
          "text": "An in-depth look at creating applications
            with XML.",
        },
        "description_html": Object {
          "cdata": "<p>An in-depth look at creating applications
            with XML.</p>",
        },
        "genre": Object {
          "text": "Computer",
        },
        "id": "bk101",
        "internal": Object {
          "contentDigest": "contentDigest",
          "type": "NodeNameXml",
        },
        "parent": "whatever",
        "price": Object {
          "text": "44.95",
        },
        "publish_date": Object {
          "text": "2000-10-01",
        },
        "title": Object {
          "text": "XML Developer's Guide",
        },
      },
      "parent": Object {
        "children": Array [],
        "content": "
      <?xml version=\\"1.0\\"?>
      <catalog>
        <book id=\\"bk101\\">
            <author>Gambardella, Matthew</author>
            <title>XML Developer's Guide</title>
            <genre>Computer</genre>
            <price>44.95</price>
            <publish_date>2000-10-01</publish_date>
            <description>An in-depth look at creating applications
            with XML.</description>
            <description_html><![CDATA[
              <p>An in-depth look at creating applications
            with XML.</p>
            ]]></description_html>
         </book>
         <book id=\\"bk102\\">
            <author>Ralls, Kim</author>
            <title>Midnight Rain</title>
            <genre>Fantasy</genre>
            <price>5.95</price>
            <publish_date>2000-12-16</publish_date>
            <description>A former architect battles corporate zombies,
            an evil sorceress, and her own childhood to become queen
            of the world.</description>
            <description_html><![CDATA[
              <p>A former architect battles <strong>corporate zombies</strong>,
            an evil sorceress, and her own childhood to become queen
            of the world.</p>
            ]]></description_html>
         </book>
      </catalog>
    ",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
          "mediaType": "application/xml",
          "name": "test",
        },
        "name": "nodeName",
        "parent": null,
      },
    },
  ],
  Array [
    Object {
      "child": Object {
        "attributes": Object {
          "id": "bk102",
        },
        "author": Object {
          "text": "Ralls, Kim",
        },
        "children": Array [],
        "description": Object {
          "text": "A former architect battles corporate zombies,
            an evil sorceress, and her own childhood to become queen
            of the world.",
        },
        "description_html": Object {
          "cdata": "<p>A former architect battles <strong>corporate zombies</strong>,
            an evil sorceress, and her own childhood to become queen
            of the world.</p>",
        },
        "genre": Object {
          "text": "Fantasy",
        },
        "id": "bk102",
        "internal": Object {
          "contentDigest": "contentDigest",
          "type": "NodeNameXml",
        },
        "parent": "whatever",
        "price": Object {
          "text": "5.95",
        },
        "publish_date": Object {
          "text": "2000-12-16",
        },
        "title": Object {
          "text": "Midnight Rain",
        },
      },
      "parent": Object {
        "children": Array [],
        "content": "
      <?xml version=\\"1.0\\"?>
      <catalog>
        <book id=\\"bk101\\">
            <author>Gambardella, Matthew</author>
            <title>XML Developer's Guide</title>
            <genre>Computer</genre>
            <price>44.95</price>
            <publish_date>2000-10-01</publish_date>
            <description>An in-depth look at creating applications
            with XML.</description>
            <description_html><![CDATA[
              <p>An in-depth look at creating applications
            with XML.</p>
            ]]></description_html>
         </book>
         <book id=\\"bk102\\">
            <author>Ralls, Kim</author>
            <title>Midnight Rain</title>
            <genre>Fantasy</genre>
            <price>5.95</price>
            <publish_date>2000-12-16</publish_date>
            <description>A former architect battles corporate zombies,
            an evil sorceress, and her own childhood to become queen
            of the world.</description>
            <description_html><![CDATA[
              <p>A former architect battles <strong>corporate zombies</strong>,
            an evil sorceress, and her own childhood to become queen
            of the world.</p>
            ]]></description_html>
         </book>
      </catalog>
    ",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
          "mediaType": "application/xml",
          "name": "test",
        },
        "name": "nodeName",
        "parent": null,
      },
    },
  ],
]
`;
